### DISCUSSION TITLE
How small of a machine would be too much?

### DISCUSSION THREAD

**Discussion by fenilli** [statelyai/xstate#2987] [Q&A] [answered]
Like, I'm thinking of making many machines because almost any part of an UI could be translated to ftm and statecharts, but how much of a bottleneck that would be?

Like making input machine, toggle machine and so on, and using those in bigger machines afterwards, like form machines, datatable machines.

**Comment by mattpocock** [comment, marked as answer]
At its most basic level, you can think of a state machine as a reducer, which stores a bit of data in memory. We haven't heard reports from users using _lots_ of state machines in an app, so I think you're free to model as many as you'd like.
